### PR TITLE
fix: [PL-31886]: Add MinMaxSlider component to UI Core library

### PR DESCRIPTION
**Pull request by praneshg239**
### Use case in SRM module inside CD's verify step
Pasted Graphic 11

### Storybook screenshot
image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Review by tan-nhu — APPROVED** [review]

**Review by satyam-harness — DISMISSED** [review]

File: packages/uicore/src/components/MinMaxSlider/MinMaxSlider.css [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  border: none;
+  border-radius: 50%;
+  box-shadow: 0 0 1px 1px var(--grey-200);
+  cursor: pointer;
+  height: 18px;
+  width: 18px;
+  margin-top: 4px;
+  pointer-events: all;
+  position: relative;
+}
```
Comment (abhinavrastogi-harness) [inline packages/uicore/src/components/MinMaxSlider/MinMaxSlider.css:94]: why do we need browser specific css? All properties seem to be the same.
We also already have a webpack plugin which auto-prefixes cross-browser css.

Comment (praneshg239) [reply packages/uicore/src/components/MinMaxSlider/MinMaxSlider.css:94]: same I thought initially and while I tested in Mozilla with `-webkit-slider-thumb` alone, it was breaking the styles (as below).
image

Comment (praneshg239) [reply packages/uicore/src/components/MinMaxSlider/MinMaxSlider.css:94]: Also the thumb was not visible when I combined both styles together. So I separated them and using the same styles for both.

File: packages/uicore/src/components/MinMaxSlider/MinMaxSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+  // Get min and max values when their state changes
+  useEffect(() => {
+    // To prevent calling onChange during mount
+    if (isMounted.current) {
+      delayedOnChange({ min: minVal, max: maxVal })
+    } else {
+      isMounted.current = true
+    }
+  }, [minVal, maxVal, delayedOnChange])
```
[the remark this answers is not in the document]
Comment (praneshg239) [reply packages/uicore/src/components/MinMaxSlider/MinMaxSlider.tsx]: Updated @rameshx , please check

**Review by rameshx — DISMISSED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by rameshx — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**